public/app/plugins/datasource/loki/modifyQuery.ts

```diff
@@ -90,6 +90,33 @@ export function addLabelFormatToQuery(query: string, labelFormat: { originalLabe
   return addLabelFormat(query, logQueryPositions, labelFormat);
 }
 
+/**
+ * Removes all comments from query.
+ * It uses  LogQL parser to find all LineComments and removes them.
+ *
+ * @param query
+ */
+export function removeCommentsFromQuery(query: string): string {
+  const lineCommentPositions = getLineCommentPositions(query);
+
+  if (!lineCommentPositions.length) {
+    return query;
+  }
+
+  let newQuery = '';
+  let prev = 0;
+
+  for (let i = 0; i < lineCommentPositions.length; i++) {
```

Comment:
How about:

```suggestion
  for (let match of lineCommentPositions) {
```

Reply:
Yeah, also great suggestion. I have changed it to `lineCommentPosition of lineCommentPositions`  inhttps://github.com/grafana/grafana/commit/cdf0c839e3e48fa18afb1b0cd7a04a852bfeaab4